src/test/java/au/org/ala/biocache/dao/FilterQueryParserTest.java

```diff
@@ -219,7 +221,7 @@ private void runFqParsingTest(List<Facet> facets) {
     public void testActiveFacetObj_invalidfq() {
         SpatialSearchRequestParams query = new SpatialSearchRequestParams();
         // Construct fq
-        query.setFq(new String[] {null, "", " ", "month", "   month  ", "(month", "month)", "(month:\"11\"", "month:\"11\" )", "month\"11\"", ":\"11\"", "    :\"11\"", "(:\"11\")", "(    :\"11\")", "month:", "month:   ",  "(month:   )", "-(month:   )"});
+        query.setFq(new String[] {null, "", " ", "month", "   month  ", "(month", "month)", "(month:\"11\"", "month\"11\"", ":\"11\"", "    :\"11\"", "(:\"11\")", "(    :\"11\")", "month:", "month:   ",  "(month:   )", "-(month:   )"});
```

Comment:
no need to remove test for invalid fq `"month:\"11\" )"` if XOR test used

Reply:
Hi Bruce, month:"11") should be removed from the test case. Because a ')' in the value could be valid. Like `-license:"CC-BY-NC 4.0 (Int)"`

The mistake I made in the original code is, I always treat () as only appearing at the start/end as a wrapping of the whole statement like (xxx:xxx OR yyy:yyy) but didn't realize that () can appear inside a value like xxx:xxx OR yyy:y(yy)

So the if statement in line 218 only checks those starts with -( or ( but not ending with ), these are definitely incorrect. In this case null is returned.

`(fq.startsWith("-(") || fq.startsWith("(")) ^ fq.endsWith(")")` returns null for `-license:"CC-BY-NC" AND -license:"CC-BY-NC 4.0 (Int)"` so it's not correct here.